source/templates.tex

```diff
@@ -7260,32 +7261,32 @@
 \end{codeblock}
 \end{example}
 
-When all template arguments have been deduced or obtained from
-default template arguments, all uses of template parameters in
-the template parameter list of the template and the function type
-are replaced with the corresponding deduced
-or default argument values. If the substitution results in an
-invalid type, as described above, type deduction fails.
-If the function template has associated constraints\iref{temp.constr.decl},
-those constraints are checked for satisfaction\iref{temp.constr.constr}.
-If the constraints are not satisfied, type deduction fails.
+The template argument values determined above by template argument deduction
+or default template arguments are substituted for their corresponding template
```

Comment:
I dislike "determined above by ... or default template arguments". We do not determined them "by default template arguments". We obtain them, which is what the original text says.

Maybe just insert "obtained from" before "default template arguments", or maybe:

> The template argument values found by template argument deduction or obtained from default template arguments are substituted ...

Also, "template argument **values**" ? These are not necessarily values.